frontend/apps/monastery/views/api400-designer/model/algos.mjs

```diff
@@ -0,0 +1,585 @@
+    }
+    else return `SCR NAME()   READ()`;
+};
+
+/**
+ * Convert the SCR KEYS node into apicl command
+ * SCR KEYS : Screen Scrapping , hit the command and keys on particular coordinates
+ * @param node  Command Obect for SCR KEYS
+ * @returns SCR KEYS apicl command
+ */
+const _convertForScrkeys = function (node) {
+    let keysVariables = [];
+    let listBoxValues = JSON.parse(node.listbox);
+    if (node.listbox) {
+        if (listBoxValues && listBoxValues.length > 0)
+            for (const scrPropertiesObj of listBoxValues) {
+                if (scrPropertiesObj.some(value => value != "")) keysVariables.push(`${scrPropertiesObj[0].trim()},${scrPropertiesObj[1].trim()},${scrPropertiesObj[2].trim()}`);
+            }
+        return `SCR NAME(${node.session ? node.session : ''})   KEYS(${keysVariables.join(" : ")})`;
+    }
+    else return `SCR NAME()   KEYS()`
+};
+
+/**
+ * Convert the SCR OPS node into apicl command
+ * SCR OPS : Screen Scrapping , start , release and stop the session
+ * @param node  Command Obect for SCR OPS
+ * @returns SCR OPS apicl command
+ */
+const _convertForScrops = function (node) {
+    return `SCR NAME(${node.session ? node.session : ''}) ${node.scrops ? node.scrops.toUpperCase() : ''}`;
+};
+
+/**
+ * Convert the RUNJS MOD node into apicl command
+ * RUNJS MOD : Execute the Javascript Code
+ * @param node  Command Obect for RUNJS MOD
+ * @returns RUNJS MOD apicl command
+ */
+const _convertForMod = function (node) {
+    return `RUNJS MOD(${node.modulename || ''})`;
+};
+
+/**
+ * Convert the SUBSTR node into apicl command
+ * SUBSTR : To Extract out the Substring from a String
+ * @param node  Command Obect for SUBSTR
+ * @returns SUBSTR apicl command
+ */
+const _convertForSubstr = function (node) {
+    return `CHGVAR  VAR(${node.variable ? node.variable.trim() : ''})  VALUE(SUBSTR DO(${node.string ? node.string.trim() : ''}:${node.index ? node.index.trim() : ''}:${node.noofchar ? node.noofchar.trim() : ''}))`
+};
+
+/**
+ * Convert the ENDAPI node into apicl command
+ * ENDAPI : To End the API
+ * @param node  Command Obect for ENDAPI
+ * @returns ENDAPI apicl command
+ */
+const _convertForEndapi = function () {
+    return `ENDAPI`;
+};
+
+/**
+ * To check next command node id, particularly in GOTO case
+ * @param {*} nodeObj Command Object
+ * @param {*} nodes All the Command's Object
+ * @returns 
+ */
+const _saveNextNodeIdsInFlow = function (nodeObj, nodes) {
+    let nextNodeObj = checkNodeInAllNodes(nodeObj, nodes);
+    if (nextNodeObj && nextNodeObj.id && nextNodeObj.nodeName != 'condition') {
+        nodeToAddLater.push(nextNodeObj.id);
+        _saveNextNodeIdsInFlow(nextNodeObj, nodes);
+    } else {
+        return;
+    }
+};
+
+/**
+ * Fecth the node which was included in the dependencies of other node
+ * @param {*} node Command Object
+ * @param {*} allnodes All the Commmand's Object
+ * @returns found node object
+ */
+const checkNodeInAllNodes = function (node, allnodes) {
```

Comment:
If this is internal then add _ at the start, else move up to public functions. Don't mix non-public and public functions.

Reply:
We are callling/using this function from 2 different files, that is why we have not put the _

Reply:
Then move it amongst public functions - the org of code should be

- imports / requires
- constants
- public functions
- private functions
- export declaration at the end

Reply:
Ok , we will change it. Thanks